backend/benefit/terms/models.py

```diff
@@ -0,0 +1,135 @@
+from django.utils.translation import gettext_lazy as _
+from terms.enums import TermsType
+
+from shared.models.abstract_models import TimeStampedModel, UUIDModel
+
+
+class Terms(UUIDModel, TimeStampedModel):
+    terms_type = models.CharField(
+        max_length=64,
+        verbose_name=_("type of terms"),
+        choices=TermsType.choices,
+        default=TermsType.APPLICANT_TERMS,
+    )
+
+    """
+    If effective_from is set to null, that means the terms are not to be displayed to the applicant.
+    """
+    effective_from = models.DateField(
+        verbose_name=_("first day these terms are in effect"), null=True, blank=True
+    )
+
+    terms_pdf_fi = models.FileField(verbose_name=_("finnish terms (pdf file)"))
+    terms_pdf_en = models.FileField(verbose_name=_("english terms (pdf file)"))
+    terms_pdf_sv = models.FileField(verbose_name=_("swedish terms (pdf file)"))
+
+    @property
+    def is_editable(self):
+        return self.effective_from is None or self.effective_from > date.today()
+
+    def __str__(self):
+        return (
+            f"{self.terms_type}-{self.effective_from if self.effective_from else 'NA'}"
+        )
+
+    class Meta:
+        db_table = "bf_terms"
+        verbose_name = _("terms")
+        verbose_name_plural = _("terms")
+        unique_together = ("terms_type", "effective_from")
+
+
+class ApplicantConsent(UUIDModel, TimeStampedModel):
+    """
+    This model represents a consent that the user gives when approving Terms.
+    The consent is represented as a checkbox+text in the UI.
+    """
+
+    terms = models.ForeignKey(
+        Terms,
+        verbose_name=_("terms"),
+        related_name="applicant_consents",
+        on_delete=models.CASCADE,
+    )
+    text_fi = models.CharField(
+        max_length=256, verbose_name=_("finnish text for the consent checkbox")
+    )
+    text_en = models.CharField(
+        max_length=256, verbose_name=_("english text for the consent checkbox")
+    )
+    text_sv = models.CharField(
+        max_length=256, verbose_name=_("swedish text for the consent checkbox")
+    )
+
+    def __str__(self):
+        return f"{self.terms} consent: {self.text_fi}"
+
+    class Meta:
+        db_table = "bf_applicationconsents"
+        verbose_name = _("application consent")
+        verbose_name_plural = _("application consents")
+
+
+class AbstractTermsApproval(UUIDModel, TimeStampedModel):
+    approved_at = models.DateTimeField(verbose_name=_("timestamp of approval"))
+    approved_by = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        verbose_name=_("user who approved the terms"),
+    )
+    terms = models.ForeignKey(
+        Terms,
+        verbose_name=_("terms"),
+        on_delete=models.CASCADE,
+    )
+    """
+    API validates that the selected_applicant_consents must belong to the terms of this approval
+    """
+    selected_applicant_consents = models.ManyToManyField(
+        ApplicantConsent,
```

Comment:
I wonder if should we add Company FK to this model as well because it's needed for the `TERMS_OF_SERVICE` terms_type